compiler/rustc_lint/src/context.rs

```diff
@@ -835,7 +835,12 @@ impl<'tcx> LateContext<'tcx> {
                     return Ok(path);
                 }
 
-                path.push(disambiguated_data.data.as_symbol());
+                path.push(match disambiguated_data.data.get_name() {
+                    DefPathDataName::Named(name) => name,
+                    DefPathDataName::Anon { namespace } => {
+                        Symbol::intern(&format!("{{{{{}}}}}", namespace))
+                    }
```

Comment:
Hmm, if you're doing this just to avoid the perf hit in the other case, you could still do `name => Symbol::intern(&name.to_string())`.

But also, ideally lints using this would just get an empty `Vec` back when the path contains anything other than simple names (see comment on `fn get_def_path`, about 100 lines above, on how it's meant to be used).

Reply:
I am not really doing this to avoid the perf hit but rather because `path` is a `Vec<Symbol>` so I need to dynamically create a symbol for the old naming scheme.
As mentioned in the PR description, I wonder if I should re-add a `DefPathData::as_legacy_symbol(&self) -> Symbol` method which would return the double braced namespace symbols?

Reply:
> so I need to dynamically create a symbol for the old naming scheme

I meant the perf hit of doing that, even in the case where you can just extract the `Symbol` from the `Named` case`.

That is, you *could* just always do `Symbol::intern(&name.to_string())`, without special-casing `Named`, and it should still work AFAICT.